### ISSUE TITLE
Cannot use custom Selenium port in Sauce Connect with wdio

### ISSUE THREAD

**Issue by klabak** [webdriverio/webdriverio#1754]
## The problem

Hi there! I'm trying to get wdio to play nicely with Sauce Connect. I've run into an issue when launching Sauce Connect with a custom Selenium port (set as the 'port' option in sauceConnectOpts. See below). I can run wdio with Sauce connect using the default ports (not having any port defined) and it connects properly (on 4445 by default). I've tried all manner of other configurations but haven't figured out how to connect wdio to the custom Selenium server out on Sauce.

## Environment

* WebdriverIO version: 4.4.0
* Node.js version: 4.2.2
* Running wdio testrunner
* Synchronous test mode
* Additional wdio packages used (if applicable): wdio-sauce-service

## Link to Selenium/WebdriverIO logs
My console logs: See that Sauce Connect launches properly and starts Selenium on the port specified. Once wdio takes over it errors out saying it can't find it.

Opening local tunnel using Sauce Connect
Starting sc with args: -u XXXXXXXX -k XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXX --tunnel-identifier testTunnel -P 4400 --readyfile /var/folders/0v/d6j5wj0x3x3c7sn7tv9ns_p83dmt4q/T/sc-launcher-readyfile
06 Dec 17:20:23 - Sauce Connect 4.3.16, build 2399 c7e5fec
06 Dec 17:20:23 - Starting up; pid 21610
06 Dec 17:20:23 - Command line arguments: /node_modules/wdio-sauce-service/node_modules/sauce-connect-launcher/sc/sc-4.3.16-osx/bin/sc -u myUserId -k **** --tunnel-identifier testTunnel -P 4400 --readyfile /var/folders/0v/d6j5wj0x3x3c7sn7tv9ns_p83dmt4q/T/sc-launcher-readyfile
06 Dec 17:20:23 - Log file: /tmp/sc_testTunnel.log
06 Dec 17:20:23 - Pid file: /tmp/sc_client_testTunnel.pid
06 Dec 17:20:23 - Timezone: EST GMT offset: -5h
06 Dec 17:20:23 - Using no proxy for connecting to Sauce Labs REST API.
06 Dec 17:20:23 - Resolving saucelabs.com to 162.222.75.243 took 0 ms.
06 Dec 17:20:23 - ***********************************************************
06 Dec 17:20:23 - A newer version of Sauce Connect (build 3147) is available!
06 Dec 17:20:23 -
Download it here:
06 Dec 17:20:23 - https://saucelabs.com/downloads/sc-4.4.2-osx.zip
06 Dec 17:20:23 - ***********************************************************
06 Dec 17:20:23 - Started scproxy on port 51373.
06 Dec 17:20:23 - Please wait for 'you may start your tests' to start your tests.
Creating tunnel with Sauce Labs
06 Dec 17:20:23 - Starting secure remote tunnel VM...
06 Dec 17:20:29 - Secure remote tunnel VM provisioned.
06 Dec 17:20:29 - Tunnel ID: 856a70a4744942029b215766b21dc4e3
06 Dec 17:20:30 - Secure remote tunnel VM is now: booting
06 Dec 17:20:41 - Secure remote tunnel VM is now: running
06 Dec 17:20:41 - Using no proxy for connecting to tunnel VM.
06 Dec 17:20:41 - Resolving tunnel hostname to 162.222.78.192 took 40ms.
06 Dec 17:20:41 - Starting Selenium listener...
06 Dec 17:20:41 - Establishing secure TLS connection to tunnel...
06 Dec 17:20:41 - Selenium listener started on port 4400.
06 Dec 17:20:45 - Sauce Connect is up, you may start your tests.
Detected sc ready
Testing tunnel ready

Selenium 2.0 / webdriver protocol bindings implementation with helper commands in nodejs.
For a complete list of commands, visit http://webdriver.io/api.html.

[17:20:48]  COMMAND	POST 	 "/wd/hub/session"
[17:20:48]  DATA		{"desiredCapabilities":{"javascriptEnabled":true,"locationContextEnabled":true,"handlesAlerts":true,"rotatable":true,"platform":"OS X 10.9","browserName":"chrome","tunnelIdentifier":"testTunnel","name":"testJob","idleTimeout":"120","screenResolution":"1600x1200","_":"6 more keys: [\"maxDuration\",\"breakpointSize\",\"shardTestFiles\",\"maxInstances\",\"loggingPrefs\",\"requestOrigins\"]"}}
ERROR: Couldn't connect to selenium server
chrome.osx10_9
Error: Couldn't connect to selenium server

06 Dec 17:20:48 - Got signal 15
06 Dec 17:20:48 - Cleaning up.
06 Dec 17:20:48 - Removing tunnel 856a70a4744942029b215766b21dc4e3.
06 Dec 17:20:53 - All jobs using tunnel have finished.
06 Dec 17:20:53 - Waiting for the connection to terminate...
06 Dec 17:20:54 - Connection closed (8).
06 Dec 17:20:54 - Goodbye.
Closing Sauce Connect Tunnel

## Code To Reproduce Issue

Here are the options I'm using for Sauce Connect in my wdio.config.js:

`services: ['sauce'],
protocol: 'http',
    user: 'myUser',
    key: process.env.SAUCE_KEY,
    sauceConnect: true,
    sauceConnectOpts: {
        tunnelIdentifier: process.env.ENV_TUNNEL_NAME,
        verbose         : true,
        logger          : console.log,
        port            : 4400
    },`

I've tried using the wdio Selenium options along with the above as well (or not specifying them at all) and I'm getting the same error:
`host:'127.0.0.1', port: 4400, path: '/wd/hub'` 

I've tried using localhost for the host option as well.

My capabilities object (which works fine if I'm not setting the custom port so pretty sure it's not this):
`capabilities: [{
        platform : 'OS X 10.9',
        browserName     : 'chrome',
        tunnelIdentifier: process.env.ENV_TUNNEL_NAME,
        name            : process.env.SAUCE_JOB_NAME,
        idleTimeout: '120',
        screenResolution : '1600x1200',
        maxDuration: 10800,
        breakpointSize: 'xlarge',
        shardTestFiles: true,
        maxInstances: 4
    }],`

Hope I'm doing something silly here! I'm loving wdio so far and this is the last thing I need to get working before we can switch our test framework over! Any help would be greatly appreciated! Thanks!

**Comment by christian-bromann** [comment]
Duplicate of #1683
